### DISCUSSION TITLE
Neovim Error TreeSitter

### DISCUSSION THREAD

**Discussion by ItsElijahWood** [community/community#136490] [Programming Help]
### Body

Hi, first time setting up a Nvim config for Windows (I'm a linux user) I'm getting an error for treesitter. I've tried updating parsers, deleting parsers, updating lazy, re-installing treesitter but none of these worked.

Error executing vim.schedule lua callback: ...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:112: Failed to load parser for language 'lua': uv_dlopen: no error                                                                                                   stack traceback:                                                                                                                              [C]: in function '_ts_add_language'                                                                                                   ...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:112: in function 'add'                                                    ...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:111: in function 'new'                                                    ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:41: in function '_create_parser'                                          ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:108: in function 'get_parser'                                             ...a/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:141: in function 'ts_highlighter'                                         ...a/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:119: in function 'highlighter'                                            ...scope.nvim/lua/telescope/previewers/buffer_previewer.lua:248: in function 'fn'                                                     vim/_editor.lua:351: in function <vim/_editor.lua:350>

return {
    "nvim-treesitter/nvim-treesitter",
    build = ":TSUpdate",
    config = function()
        require("nvim-treesitter.configs").setup({
            -- A list of parser names, or "all"
            ensure_installed = {
                "vimdoc", "javascript", "typescript", "c", "lua", "rust",
                "jsdoc", "bash", "java", "html", "css", "zig",
            },

            -- Install parsers synchronously (only applied to `ensure_installed`)
            sync_install = false,

            -- Automatically install missing parsers when entering buffer
            -- Recommendation: set to false if you don't have `tree-sitter` CLI installed locally
            auto_install = true,

            indent = {
                enable = true
            },

            highlight = {
                -- `false` will disable the whole extension
                enable = true,

                -- Setting this to true will run `:h syntax` and tree-sitter at the same time.
                -- Set this to `true` if you depend on "syntax" being enabled (like for indentation).
                -- Using this option may slow down your editor, and you may see some duplicate highlights.
                -- Instead of true it can also be a list of languages
                additional_vim_regex_highlighting = { "markdown" },
            },
        })
    end
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Sin-cy** [comment]
Ive had this error message come up to me once. And this could either be an issue of something else interferring with the nvim tree-sitter.
In my case. This error was showing up because I was calling the plugin twice in two different config files meaning you have tree sitter running some sort function in some other plugin.lua file "example for me, in mason.lua file I have another function that is somehow calling mason function and its doing the same one in the current file as well"

From your treesitter config, It looks okay. The only thing I can think of is the lua parser not being installed causing it to not be found. 

Also what does it say when you tried updating it 
`:TSUpdate` or `:TSUpdate lua` or `:TSInstall lua` 
does the error stay the same ?

It could be a loading issue ,try adding 
`event = { "BufReadPre", "BufNewFile" },`
before the `build = ":TSUpdate",`


maybe try setting `sync_install = true `

You could also check 
/usr/local/lib/nvim 
and see it if contains the parsers 

Another thing you could try to do is rebuilt neovim but I understand thats probably not the option. But sometime it does fix the issues because the built from the startup was arranged in the wrong way. 

I would also take a look at this config file or this function
...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:112: in function 'add' 
and see whats causing the function to break and how its related to treesitter. Because its usually the top most one that breaks and then breaks all the other stuff as well.

If YOU REALLY CANT figure it out or if this does not fix the issue. 
The last way would be to either go to Stackoverflow or Reddit and paste the same question in there. There is a lot more advanced vim chads in there. More chances to find the solution.
